bundles/tools.mdsd.library.standalone.initialization/src/tools/mdsd/library/standalone/initialization/core/EclipseProjectScanner.java

```diff
@@ -0,0 +1,133 @@
+     *            Jar-Manifests.
+     */
+    public EclipseProjectScanner(Path basePath) {
+        this.basePath = basePath;
+    }
+
+    @Override
+    public void initilizationWithoutPlatform() throws StandaloneInitializationException {
+        Map<String, File> projects = Collections.emptyMap();
+        try {
+            projects = findProjects(basePath);
+        } catch (IOException e) {
+            throw new StandaloneInitializationException("Errors reading project artifacts", e);
+        }
+
+        if (projects.isEmpty())
+            throw new StandaloneInitializationException(
+                    "Could not find the eclipse project artifacts in " + basePath.toString() + ".");
+
+        for (var project : projects.entrySet()) {
+            (new ProjectURIByLocationRegistration(project.getValue(), project.getKey())).init();
+        }
+    }
+
+    protected static Map<String, File> findProjects(Path basePath) throws IOException {
+        var projectFilePath = Paths.get(PROJECT_FILE_NAME);
+
+        var res = Files
+            .find(basePath, DEFAULT_MAX_DEPTH + projectFilePath.getNameCount(),
+                    (path, attr) -> path.endsWith(projectFilePath))
+            .collect(Collectors.toMap(EclipseProjectScanner::readProjectNameFromProjectFile, path -> {
+                var result = path;
+                // Using subpath removes the root, therefore we loop manually
+                for (int i = 0; i < projectFilePath.getNameCount(); i++) {
+                    result = result.getParent();
+                }
+                return result.toFile();
+            }));
+
+        var manifestPath = Paths.get(JarFile.MANIFEST_NAME);
+        res.putAll(Files
+            .find(basePath, DEFAULT_MAX_DEPTH + manifestPath.getNameCount(),
+                    (path, attr) -> path.endsWith(Paths.get(JarFile.MANIFEST_NAME)))
+            .collect(Collectors.toMap(EclipseProjectScanner::readProjectNameFromManifestFile, path -> {
+                var result = path;
+                // Using subpath removes the root, therefore we loop manually
+                for (int i = 0; i < manifestPath.getNameCount(); i++) {
+                    result = result.getParent();
+                }
+                return result.toFile();
+            }, (a, b) -> b)));
+
+        return res;
+    }
+
+    protected static String readProjectNameFromProjectFile(Path path) {
+        Document document;
+        try {
+            document = DocumentBuilderFactory.newInstance()
+                .newDocumentBuilder()
+                .parse(Files.newInputStream(path, StandardOpenOption.READ));
+            return document.getDocumentElement()
+                .getElementsByTagName("name")
+                .item(0)
+                .getTextContent();
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            throw new RuntimeException("Error reading the project file " + path.toString(), e);
```

Comment:
Why is there a need for RuntimeException instances? The signature of the task provides means for emitting a checked exception.

Reply:
Same as above